File: src/python/core/domain/configuration/vector_metadata_config.py
from dataclasses import dataclass
from typing import List, Optional


@dataclass(frozen=True)
class VectorMetadataConfig:
    FIELD_TYPE: str = "type"
    FIELD_CANDIDATE_ID: str = "candidate_id"
    FIELD_FULLNAME: str = "fullname"
    FIELD_ENGLISH_LEVEL: str = "english_level"
    FIELD_ENGLISH_LEVEL_NUM: str = "english_level_num"
    FIELD_SENIORITY_LEVEL: str = "seniority_level"
    FIELD_YEARS_EXPERIENCE: str = "years_experience"
    FIELD_RELEVANT_YEARS: str = "relevant_years"
    FIELD_MAIN_INDUSTRY: str = "main_industry"
    FIELD_PRIMARY_SKILLS: str = "primary_skills"
    FIELD_SKILL_NAME: str = "skill_name"
    FIELD_SKILL_LEVEL: str = "skill_level"

    TYPE_CANDIDATE: str = "candidate"
    TYPE_SKILL: str = "skill"
    TYPE_LLM_INSTRUCTION: str = "llm_instruction"
    
    primary_skills_max_count: int = 5
    strong_skill_levels: List[str] = None
    
    def __post_init__(self):
        if self.strong_skill_levels is None:
            object.__setattr__(self, 'strong_skill_levels', ["High", "Very High"])


DEFAULT_VECTOR_METADATA_CONFIG = VectorMetadataConfig()

File: src/dotnet/Core/Domain/Configuration/VectorMetadataConfig.cs
namespace Rag.Candidates.Core.Domain.Configuration;

public sealed record VectorMetadataConfig
{
    public string FieldType { get; init; } = "type";
    public string FieldCandidateId { get; init; } = "candidate_id";
    public string FieldFullname { get; init; } = "fullname";
    public string FieldEnglishLevel { get; init; } = "english_level";
    public string FieldEnglishLevelNum { get; init; } = "english_level_num";
    public string FieldSeniorityLevel { get; init; } = "seniority_level";
    public string FieldYearsExperience { get; init; } = "years_experience";
    public string FieldRelevantYears { get; init; } = "relevant_years";
    public string FieldMainIndustry { get; init; } = "main_industry";
    public string FieldPrimarySkills { get; init; } = "primary_skills";
    public string FieldSkillName { get; init; } = "skill_name";
    public string FieldSkillLevel { get; init; } = "skill_level";

    public string TypeCandidate { get; init; } = "candidate";
    public string TypeSkill { get; init; } = "skill";
    public string TypeLlmInstruction { get; init; } = "llm_instruction";

    public int PrimarySkillsMaxCount { get; init; } = 5;
    public string[] StrongSkillLevels { get; init; } = new[] { "High", "Very High" };

    public static VectorMetadataConfig Default { get; } = new();
}
